packages/nestjs-authorization/src/auth-action.guard.ts

```diff
@@ -5,21 +5,59 @@ import {
   Injectable,
   mixin,
 } from '@nestjs/common';
+import { isFunctionalCondition } from './condition';
 import { Action } from './action';
 import { PermissionProvider } from './permission.provider';
+import { ConditionService } from './condition.service';
 
 export const AuthActionGuard = (actions: Action[]) => {
   @Injectable()
   class AuthActionGuardImpl implements CanActivate {
     constructor(
       @Inject('PERMISSION_PROVIDER')
-      private permissionProvider: PermissionProvider
+      private permissionProvider: PermissionProvider,
+      private conditionService: ConditionService
     ) {}
 
     async canActivate(context: ExecutionContext): Promise<boolean> {
       const req = context.switchToHttp().getRequest();
       const permissionSet = await this.permissionProvider.getPermissionSet(req);
-      return permissionSet.areAllowed(actions, context);
+      const result = actions.reduce(async (isAllowedPromise, action) => {
+        const isAllowed = await isAllowedPromise;
+        if (isAllowed) {
+          return Promise.resolve(true);
+        }
+
+        if (typeof action === 'string') {
+          return await permissionSet.areAllowed([action], context);
```

Comment:
maybe we should offer isAllowed to. Especially when i most cases only one action is checked